File: 242-valid-anagram/valid-anagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        unordered_map<int,int> mp;

        for(auto &it: s){
            mp[it]++;
        }

        for(auto &i: t){
            mp[i]--;
        }

        for(auto &i: mp){
            if(i.second!=0){
                return false;
            }
        }
        return true;
    }
};

File: 242-valid-anagram/valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        seen= {}
        if len(s) != len(t):
            return False
        for num in s:
            seen[num]=seen.get(num,0)+1

        for num in t:
            seen[num]=seen.get(num,0)-1

        for num in seen:
            if seen[num] != 0:
                return False
        return True